Dailymotion.php

```diff
@@ -334,19 +334,53 @@ public function uploadFile($filePath, $forceHostname = null, &$progressUrl = nul
         $this->timeout = null;
 
         // Upload the file to Dailymotion's servers
-        $result = json_decode(
-            $this->httpRequest($result['upload_url'], array('file' => $this->getCurlFile($filePath))),
-            true
-        );
+        if($workers === 0)
+        {
+            $result = json_decode(
+                $this->httpRequest($result['upload_url'], array('file' => $this->getCurlFile($filePath))),
+                true
+            );
+        }
+        else
+        {
+            $upload = new XUpload(
+                $result['upload_url'],
+                $filePath,
+                intval(min(max($workers,1),8)),
+                $progressCallback,
+                array('socket' => $this->proxy, 'credentials' => $this->proxyCredentials)
+            );
+            $result = $upload->start();
+        }
+
         $this->timeout = $timeout;
 
         if (isset($result['error']))
         {
             throw new DailymotionApiException($result['error']['message'], $result['error']['code']);
```

Comment:
```
-            throw new DailymotionApiException($result['error']['message'], $result['error']['code']);
+            $message = isset($result['error']['message']) ? $result['error']['message'] : $result['error'];
+            $code    = isset($result['error']['code'])    ? $result['error']['code']    : 0;
+            throw new DailymotionApiException($message, $code);```

Reply:
Thanks for your help, it has been updated